In [ ]:
import numpy as np

def generate_test_points():
    # 1. we define Matrix K
    K = np.array([
        [7291.67, 0.0,     639.5],
        [0.0,     7291.67, 511.5],
        [0.0,     0.0,     1.0]
    ])

    # 2. we define Camera Centers and Rotations
    R = np.eye(3)
    c0 = np.array([0, 0, 0])     # Cam 0 at origin
    c1 = np.array([1000, 0, 0])  # Cam 1 at X=1000

    # we calculate t= -R * c
    t0 = -R @ c0
    t1 = -R @ c1

    # 3. we build Projection Matrices P = K[R|t]
    P0 = K @ np.c_[R, t0]
    P1 = K @ np.c_[R, t1]

    # 4. we define the 3D Point we want to find (in homogeneous coordinates)
    X_world = np.array([1000.0, 0.0, 5000.0, 1.0])

    # 5. we project through camera 0
    cam0_proj = P0 @ X_world
    u = cam0_proj[0] / cam0_proj[2]
    v = cam0_proj[1] / cam0_proj[2]

    # 6. we project through camera 1
    cam1_proj = P1 @ X_world
    u_p = cam1_proj[0] / cam1_proj[2]
    v_p = cam1_proj[1] / cam1_proj[2]

    print(f"--- Synthetic Test Data ---")
    print(f"3D Point: {X_world[:3]}")
    print(f"U: ({u:.3f}, {v:.3f})")
    print(f"U_prime: ({u_p:.3f}, {v_p:.3f})")
    
    return (u, v), (u_p, v_p)

generate_test_points()

--- Synthetic Test Data ---
3D Point: [1000.    0. 5000.]
U: (2097.834, 511.500)
U_prime: (639.500, 511.500)


((2097.834, 511.5), (639.5, 511.5))

Add a tiny bit of noise: u_noisy = u + 0.5;.

Tests tried:
vec U(1004.0835, 511.5);
vec U_prime(274.9165, 511.5);
vec expected_result(500.0, 0.0, 10000.0);

-------------------------------------------
vec expected_result(1000.0, 0.0, 5000.0, 1.0);
vec U(2097.834, 511.500);
vec U_prime(639.500, 511.500);




